RuntimeDetour/Platforms/Runtime/DetourRuntimeNETCorePlatform.cs

```diff
@@ -1,24 +1,158 @@
+        //   in a broken state.
+        //
+
+        private static void CallGetJitGuid(IntPtr jit, int index, out Guid guid) {
+            d_getVersionIdentifier getVersionIdentifier = ReadObjectVTable(jit, index)
+                .AsDelegate<d_getVersionIdentifier>();
+            getVersionIdentifier(jit, out guid);
+        }
+
+        // FIXME: .NET 5 has this method at index 2; how do we identify this?
+        private const int vtableIndex_ICorJitCompiler_getVersionIdentifier = 4;
+        private const int vtableIndex_ICorJitCompiler_getVersionIdentifier_net5 = 2;
+        [UnmanagedFunctionPointer(CallingConvention.ThisCall)]
+        private delegate void d_getVersionIdentifier(
+            IntPtr thisPtr, // ICorJitCompiler*
+            out Guid versionIdentifier
+            );
+
+        protected const int vtableIndex_ICorJitCompiler_compileMethod = 0;
+
+        protected static unsafe IntPtr* GetVTableEntry(IntPtr @object, int index)
+            => (*(IntPtr**) @object) + index;
+        protected static unsafe IntPtr ReadObjectVTable(IntPtr @object, int index)
+            => *GetVTableEntry(@object, index);
+
+        protected override void DisableInlining(MethodBase method, RuntimeMethodHandle handle) {
+            // https://github.com/dotnet/runtime/blob/89965be3ad2be404dc82bd9e688d5dd2a04bcb5f/src/coreclr/src/vm/method.hpp#L178
+            // mdcNotInline = 0x2000
+            // References to RuntimeMethodHandle (CORINFO_METHOD_HANDLE) pointing to MethodDesc
+            // can be traced as far back as https://ntcore.com/files/netint_injection.htm
+
+            // FIXME: Take a very educated guess regarding the offset to m_wFlags in MethodDesc.
+        }
+
+        public static readonly Guid Core31Jit = new Guid("d609bed1-7831-49fc-bd49-b6f054dd4d46");
+        public static readonly Guid Net50p4Jit = new Guid("6ae798bf-44bd-4e8a-b8fc-dbe1d1f4029e");
+
+        protected virtual void InstallJitHooks(IntPtr jitObject) => throw new PlatformNotSupportedException();
+
+        public static DetourRuntimeNETCorePlatform Create() {
+            try {
+                IntPtr jit = GetJitObject();
+                Guid jitGuid = GetJitGuid(jit);
+
+                DetourRuntimeNETCorePlatform platform = null;
+
+                if (jitGuid == Net50p4Jit) {
+                    platform = new DetourRuntimeNET50p4Platform();
+                } else if (jitGuid == Core31Jit) {
+                    platform = new DetourRuntimeNETCore31Platform();
+                }
+                // TODO: add more known JIT GUIDs
+
+                if (platform == null)
+                    return new DetourRuntimeNETCorePlatform();
+
+                platform?.InstallJitHooks(jit);
+                return platform;
+            } catch {
+                MMDbgLog.Log("Could not get JIT information for the runtime, falling out to the version without JIT hooks");

```

Comment:
Logging the exception here might up being helpful in odd occasions in the future.

Reply:
Done.